HanksMineralEmporium/Service/AuthenticationService/ AuthenticationService.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using HanksMineralEmporium.Core.UserManagement;
+using HanksMineralEmporium.Core.UserManagement.Exception;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+
+namespace HanksMineralEmporium.Service.AuthenticationService
+{
+    
+    public class  AuthenticationService : IAuthenticationService
+    {
+        private readonly IUserManager _userManager;
+        
+    async void LoginUser(string username, string password)
+{
+    try
+    {
+         var serviceProvider = new ServiceCollection()
+                .AddIdentity<IdentityUser, IdentityRole>()
+                .AddDefaultTokenProviders()
+                .Services
+                .BuildServiceProvider();
+        // Create a UserManager
+            var userManager = serviceProvider.GetRequiredService<UserManager<IdentityUser>>();
+            var signInManager = serviceProvider.GetRequiredService<SignInManager<IdentityUser>>();
+
+            bool rememberMe = false;
+
+        // Find the user by username
+            var user = await userManager.FindByNameAsync(username);
+            
+
+        if (user != null) 
+        {
+            Console.Write("User found");
+             // Attempt to sign in the user
+             // READ not sure if password validation is handled by the result or not
+                var result = await signInManager.PasswordSignInAsync(user, password, rememberMe, lockoutOnFailure: false);
+                if (result.Succeeded)
+                {
+                    Console.WriteLine("User logged in successfully.");
+                }
+                 else
+                {
+                    Console.WriteLine("Invalid username or password.");
+                }
+                
+        }
+
+        else{throw new Core.UserManagement.Exception.UserNotFoundException("User not found");}        
+             
+    }
+    catch (Core.UserManagement.Exception.UserNotFoundException ex)
+    {
+        Console.WriteLine($"UserNotFoundException: {ex.Message}");
+
+    }
+    
+}
+async void RegisterUser(string username, string password){
+         var serviceProvider = new ServiceCollection()
+                .AddIdentity<IdentityUser, IdentityRole>()
+                .AddDefaultTokenProviders()
+                .Services
+                .BuildServiceProvider();
+        var userManager = serviceProvider.GetRequiredService<UserManager<IdentityUser>>();
+
+         var newUser = new IdentityUser
+            {
+                UserName = username,
+                PasswordHash = password
+            };
+            if(newUser.UserName.Length < 3){
```

Comment:
Why are you validating the username here? The userManager does this for you when you create a new user. The user should be created in the userManager first in order to validate the username and password.